docs/guillotine.adoc

```diff
@@ -0,0 +1,1349 @@
+
+Moving on, _Movie.jsx_ also imports some *styling* that'll be handled by webpack the same way as in <<imports-and-dependency-chunks#webpack_rules, the previous chapter>>:
+
+
+.react4xp/shared/Movie.scss:
+[source,sass,options="nowrap"]
+----
+html, body {
+  margin: 0; padding: 0;
+}
+
+.infoContainer {
+  flex-grow: 1; flex-basis: content; padding: 0; margin: 0;
+
+  * {
+    font-family: 'DejaVu Sans', Arial, Helvetica, sans-serif; color: #444;
+  }
+
+  h2, h3 {
+    padding: 0; margin: 0; color: #0c0c0c;
+  }
+
+  h2 {
+    font-size: 34px;
+  }
+
+  p {
+    padding: 0; margin: 10px 0 0 0;
+  }
+}
+
+.info {
+  margin: 0; padding: 30px 0 0 0;
+}
+
+
+.movie {
+  margin: 0; padding: 30px; box-sizing: border-box; width: 100%; display: flex; flex-flow: row nowrap; justify-content: flex-start; align-items: flex-start;
+}
+
+.poster {
+  width: 300px; max-width: 30%; margin-right: 30px; flex: 0 1 auto;
+}
+
+.cast {
+  list-style-type: none; margin: 0; padding: 0;
+
+  .actor {
+    width: 100%; padding: 0; margin: 10px 0 0 0;
+  }
+}
+
+----
+
+{zwsp} +
+
+[[controller_mapping]]
+=== Controller mapping
+
+Here comes a little variation: in this example, we want to connect a movie content item to with the rendering of the _Movie.jsx_ entry. But we don't want to mess around with setting up a <<pages-parts-and-regions#page_template_setup, template with a part>> the way we've done so far. Instead, we can use a link:https://developer.enonic.com/docs/xp/stable/cms/mappings[controller mapping] to make that connection in code.
+
+Let's open _site.xml_ and add a mapping:
+
+.site/site.xml:
+[source,xml,options="nowrap"]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<site>
+  <form/>
+    <mappings>
+
+        <!-- Add this... -->
+        <mapping controller="/controllers/previewMovie.js" order="50">
+            <match>type:'com.enonic.app.react4xp:movie'</match>
+        </mapping>
+        <!-- ...and that's it. -->
+
+    </mappings>
+</site>
+----
+
+Now, every _movie_ content item in Content Studio is always rendered with a particular controller: _/controllers/previewMovie.js_.
+
+Two important points when using a controller mapping like this:
+
+[NOTE]
+====
+First, the controller reference in a mapping in _site.xml_ must always refer to *the runtime name of the controller*. In our case, the source file of our controller is _/controllers/previewMovie_ *_.es6_*, but at compile time, this is compiled into *_.js_* which is used at XP runtime.
+
+Second, controller mappings use qualified content type names that have *the name of the app* in it: `com.enonic.app.react4xp`. If/when you use a different name for your app, make sure to update content type references like this, e.g. `<match>type:'my.awesome.app:movie'</match>`
```

Comment:
This could maybe even be part of the main text? I assume everyone will not use the default name in production?

Reply:
Hmm, isn't notes supposed to be for emphasis, "above" the main text? Or are they more like side notes, "feel free to skip this"? Hope I don't have to rewrite too many of them...

Reply:
Differs from context. Its fine  if we keep it like this. The most important is mentioning it.

Reply:
Okay, I'll just keep it then.